subsys/bluetooth/services/ots/ots_client.c

```diff
@@ -1185,6 +1288,35 @@ int bt_ots_client_read_object_data(struct bt_ots_client *otc_inst,
 	return oacp_read(conn, inst);
 }
 
+int bt_ots_client_write_object_data(struct bt_ots_client *otc_inst,
+				   struct bt_conn *conn, uint8_t *buf,size_t len,
+			   off_t offset, uint8_t mode)
+{
+	struct bt_otc_internal_instance_t *inst;
+
+	if (!conn) {
+		BT_WARN("Invalid Connection");
+		return -ENOTCONN;
+	} else if (!otc_inst) {
+		BT_ERR("Invalid OTC instance");
+		return -EINVAL;
+	}
```

Comment:
Consider using `CHECKIF` to do parameter checks like these

Reply:
Use CHECKIF to validate input parameter.